### ISSUE TITLE
No reference link on nodes of DAG Dependencies

### ISSUE THREAD

**Issue by hwooson12** [apache/airflow#25861]
### Apache Airflow version

2.3.3

### What happened

On DAG Dependencies, all nodes don’t have links even though they seem clickable.

<img width="2077" alt="image" src="https://user-images.githubusercontent.com/16971553/185820935-8f067ede-336f-49f6-b9f4-27e54e8a81c0.png">

The mouse pointer turns clickable(sorry, I couldn't capture the mouse pointer) and it activates the node. However it cannot be actually clicked. 

### What you think should happen instead

Each node has its reference so that having a link to move to a page that belongs to it could be better for users in terms of UX.

### Operating System

Linux, but it's not depending on OS

### Deployment

Other

### Deployment details

Deployed with customized helm charts

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
Yes, I've wanted to add links and make this page more integrated into the app.

That's what I've done over in https://github.com/apache/airflow/pull/25707. Right now, we're only showing dataset dependencies there. But we may expand it to be a more dynamic replacement of tis view.

**Comment by bbovenzi** [comment]
In Airflow 3, you can now click between dags and assets in the dependencies graph
